### PR TITLE
vala: split valadoc to break cyclic dependency

### LINKED ISSUE #10473: New cyclic dependency
Some update reintroduced a dependency cycle: `cargo` -> `libgit2` -> `git` -> `libsecret` -> `vala` -> `graphviz` -> `librsvg` -> `cargo`
To me it seems `libgit2` depending on `git` in the host is not necessary because the former builds w/o the latter?

### PR DESCRIPTION
**Pull request by CameronNemo**
@Gottox

@Duncaen I am moving a subpackage from one srcpkg to another, will that create issues?

closes #10473

### CODE REVIEW

File: srcpkgs/valadoc/template [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Template file for 'valadoc'
```
Comment (Cogitri) [inline srcpkgs/valadoc/template]: Maybe add a header here saying that this has to be kept in sync with vala (see chromium/firefox for now it's usually done)

File: srcpkgs/vala/template [hunk 2]
```diff
@@ -15,13 +16,6 @@ distfiles="${GNOME_SITE}/vala/${version%.*}/vala-${version}.tar.xz"
...
-# for valadoc
-if [ "$CROSS_BUILD" ]; then
-	hostmakedepends+=" vala"
-else
-	make_build_args="VALAC="${XBPS_BUILDDIR}/${pkgname}-${version}/compiler/valac""
-fi
-
 pre_configure() {
 	autoreconf -if
 }
```
Comment (Cogitri) [inline srcpkgs/vala/template]: No need for this and the automake/libtool dep now that we don't patch this package anymore

**Comment by Duncaen** [comment]
Might result in lost files on update, but I guess this one is less problematic compared to other cases where it breaks other programs.
A work around which might work is to push it in two steps, first remove the `valadoc` subpackage, let it build and then push the new `valadoc` package, this way the files in each package have different `mtime`s and xbps would not see the files as obsolete when installing `valadoc` before updating `vala`.

**Comment by jnbr** [comment]
merged as suggested in 7953e010b724806915f3ed0044ab71e5a26228bd

**Pull request closed without merging**